#### About
> Entropy

Entropy is a measure of the uncertainty or randomness of a system or a random variable. In information theory, entropy is used to quantify the amount of information contained in a message or signal.

The entropy of a discrete random variable X with probability mass function p(x) is defined as:

H(X) = - Σ p(x) log p(x)

Use cases include Data compression, Cryptography, Image processing and ML

Suppose we have a discrete random variable X with the following probability mass function:

p(0) = 0.3
p(1) = 0.4
p(2) = 0.2
p(3) = 0.1

In [1]:
#calculate entropy

import math

p = [0.3, 0.4, 0.2, 0.1]

entropy = 0

for i in range(len(p)):
    if p[i] != 0:
        entropy -= p[i] * math.log(p[i], 2)

print("Entropy =", entropy)


Entropy = 1.8464393446710154


The entropy of X is approximately 1.846 bits. This means that on average, we need 1.846 bits to encode each value of X.